File: math/NumericalAnalysis/test/plotspline.m
clear;
close all;

Path='./data/';

X=csvread(strcat(Path,'x.csv'),1,0);
Y=csvread(strcat(Path,'y.csv'),1,0);
spline_X=csvread(strcat(Path,'spline_x.csv'),1,0);
spline_Y=csvread(strcat(Path,'spline_y.csv'),1,0);

figure(1)
plot(X,Y,'+');hold on;
plot(spline_X,spline_Y,'-');hold on;

File: math/NumericalAnalysis/test/post/plotspline.py
# /*
# ****************************************************************************
# * plotspline.py:
# * Illustration of results of Polynomial interpolation using spline
# *
# * by Hu.ZH(CrossOcean.ai)
# ****************************************************************************
# */


import pandas
import matplotlib.pyplot as plt
import math

path = '../data/'
X = pandas.read_csv(path+'x.csv')
Y = pandas.read_csv(path+'y.csv')
spline_X = pandas.read_csv(path+'spline_x.csv')
spline_Y = pandas.read_csv(path+'spline_y.csv')


plt.figure(1, figsize=(8, 6))
plt.suptitle("Polynomial interpolation using spline", fontsize=13)
plt.plot(X, Y, '+', lw=2, markersize=13)
plt.plot(spline_X, spline_Y, '--k', lw=2)
plt.show()
